mobile-app/app/screens/AppNavigator/screens/Balances/hooks/DexTokenPrice.ts

```diff
@@ -0,0 +1,101 @@
+import { useEffect } from 'react'
+import BigNumber from 'bignumber.js'
+import { RootState } from '@store'
+import { fetchPoolPairs } from '@store/wallet'
+import { PoolPairData } from '@defichain/whale-api-client/dist/api/poolpairs'
+import { ActivePrice } from '@defichain/whale-api-client/dist/api/prices'
+import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
+import { useDispatch, useSelector } from 'react-redux'
+import { getActivePrice } from '@screens/AppNavigator/screens/Auctions/helpers/ActivePrice'
+import { checkIfPair, findPath, GraphProps } from '@screens/AppNavigator/screens/Dex/helpers/path-finding'
+
+interface DexTokenPrice {
+  getDexTokenActivePrice: (symbol: string, activePrice?: ActivePrice) => BigNumber
+  calculatePriceRates: (fromTokenSymbol: string, pairs: PoolPairData[], amount: string) => { aToBPrice: BigNumber, bToAPrice: BigNumber, estimated: BigNumber }
+  getSelectedPoolPairs: (tokenASymbol: string, tokenBSymbol: string) => PoolPairData[]
+}
+
+export function useDexTokenPrice (): DexTokenPrice {
```

Comment:
Personally think `useTokenPrice` could be a better naming, since within this hook you have the logic to retrieve price from
- DEX (for non-loan related token) and
- from active price for loan-related token

Reply:
Make more sense, Thanks for suggestion